packages/idempotency/src/persistence/PersistenceLayer.ts

```diff
@@ -1,24 +1,161 @@
+
+  // envVarsService is always initialized in the constructor
+  private envVarsService!: EnvironmentVariablesService;
+
+  private expiresAfterSeconds: number;
+
+  private functionName: string = '';
+
+  private hashDigest: BinaryToTextEncoding;
+
+  private hashFunction: string;
+
+  public constructor() { 
+    this.setEnvVarsService();
+    this.expiresAfterSeconds = 60 * 60; //one hour is the default expiration
+    this.hashFunction = 'md5';
+    this.hashDigest = 'base64';
+        
+  }
+  public configure(functionName: string = ''): void {
+    this.functionName = this.getEnvVarsService().getLambdaFunctionName() + '.' + functionName;
+  }
+
+  /**
+   * Deletes a record from the persistence store for the persistence key generated from the data passed in.
+   * 
+   * @param data - the data payload that will be hashed to create the hash portion of the idempotency key
+   */
+  public async deleteRecord(data: unknown): Promise<void> { 
+    const idempotencyRecord: IdempotencyRecord = 
+    new IdempotencyRecord(this.getHashedIdempotencyKey(data),
+      IdempotencyRecordStatus.EXPIRED,
+      undefined,
+      undefined,
+      undefined,
+      undefined
+    );
+    
+    this._deleteRecord(idempotencyRecord);
+  }
+  /**
+   * Retrieves idempotency key for the provided data and fetches data for that key from the persistence store
+   * 
+   * @param data - the data payload that will be hashed to create the hash portion of the idempotency key
+   */
+  public async getRecord(data: unknown): Promise<IdempotencyRecord> {
+    const idempotencyKey: string = this.getHashedIdempotencyKey(data);
+
+    return this._getRecord(idempotencyKey);
+  }
+
+  /**
+   * Saves a record indicating that the function's execution is currently in progress
+   * 
+   * @param data - the data payload that will be hashed to create the hash portion of the idempotency key
+   */
+  public async saveInProgress(data: unknown): Promise<void> { 
+    const idempotencyRecord: IdempotencyRecord = 
+    new IdempotencyRecord(this.getHashedIdempotencyKey(data),
+      IdempotencyRecordStatus.INPROGRESS,
+      this.getExpiryTimestamp(),
+      undefined,
+      undefined,
+      undefined
+    );
+
+    return this._putRecord(idempotencyRecord);
+  }
+
+  /**
+   * Saves a record of the function completing successfully. This will create a record with a COMPLETED status
+   * and will save the result of the completed function in the idempotency record.
+   * 
+   * @param data - the data payload that will be hashed to create the hash portion of the idempotency key
+   * @param result - the result of the successfully completed function
+   */
+  public async saveSuccess(data: unknown, result: unknown): Promise<void> { 
+    const idempotencyRecord: IdempotencyRecord = 
+    new IdempotencyRecord(this.getHashedIdempotencyKey(data),
+      IdempotencyRecordStatus.COMPLETED,
+      this.getExpiryTimestamp(),
+      undefined,
+      JSON.stringify(result),
```

Comment:
I found this while making a fix for another issue, it seems like our types aren't matching here. The Idempotency Record response is currently defined as an object | undefined, but here you're passing in a string. It looks like we diverged a bit from the Python, where the Python converts the response to a string and puts that into a record, but in our implementation of the record it accepts the direct response object.